lib/new_relic/agent/agent_logger.rb

```diff
@@ -132,7 +132,7 @@ def create_null_logger
       end
 
       def wants_stdout?
-        ::NewRelic::Agent.config[:log_file_path].casecmp("STDOUT").zero?
+        ::NewRelic::Agent.config[:log_file_path].casecmp("STDOUT") == 0
```

Comment:
Given the micro level performance concern, we ought to stop initializing a new string `'STDOUT'` every time here and use a constant instead. NOTE: I recommend naming the constant something like `STANDARD_OUT` to avoid conflicting with Ruby's existing `STDOUT` constant.

```shell
$ ruby test.rb                                                                                                           
       user     system      total        real
zero?  0.008423   0.000190   0.008613 (  0.008612)
== 0  0.006353   0.000066   0.006419 (  0.006451)
zero? with a constant  0.004898   0.000004   0.004902 (  0.004904)
== 0 with a constant  0.004168   0.000001   0.004169 (  0.004171)
upcase and match with a constant  0.066901   0.001593   0.068494 (  0.068576)
```